### PR TITLE
Change katello version to 3.17

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
these packages were never in an actual release, should we bother bumping epoch?

**Comment by jlsherrill** [comment]
related katello pr: https://github.com/Katello/katello/pull/8874/

**Comment by ekohl** [comment]
I'm not sure we need to bump epoch since there are no support guarantees for nightly. I would clearly document it since going back in the changelog is a bit odd. It also means untagging in koji. Probably also worth a notice on discourse for users/developers who have such a box.

**Comment by jlsherrill** [comment]
@ekohl you mean a comment in the change log?

**Comment by ekohl** [comment]
Yes, or squash the changelog from 4.0. My main concern is that the changelogs are visible to users and it's surprising to go from 4.0 to 3.17 (and in the future back to 4.0). Wondering what @theforeman/packaging thinks about it.

**Comment by ehelms** [comment]
Yea, squash it down!

**Comment by ehelms** [comment]
I think there are a few dependencies you will need to update with this PR to get it to build.

**Comment by ehelms** [comment]
[test rpm]

**Comment by ehelms** [comment]
```
DEBUG util.py:598:  Error: No Package found for tfm-rubygem(pulp_rpm_client) < 3.5.0
```

**Comment by jlsherrill** [comment]
Packaging actually got ahead of katello in this case:  This pr has the changes we need  https://github.com/Katello/katello/pull/8869/

and then we can update this spec file

File: packages/katello/katello/katello.spec [hunk 1]
```diff
@@ -180,6 +180,9 @@ Provides a federation of katello services
 # the files section is empty, but without it no RPM will be generated
 
 %changelog
+* Tue Aug 04 2020 Justin Sherrill <jsherril@redhat.com> 3.17.0-0.1.master
+- change version to 3.17.0
+
 * Thu Jun 18 2020 Barnaby Court <bcourt@redhat.com> - 4.0.0-0.3.master
```
Comment (ekohl) [inline packages/katello/katello/katello.spec:186]: Can you squash this up too?

File: packages/katello/katello/katello.spec [hunk 2]
```diff
@@ -180,6 +180,9 @@ Provides a federation of katello services
 # the files section is empty, but without it no RPM will be generated
 
 %changelog
+* Tue Aug 04 2020 Justin Sherrill <jsherril@redhat.com> 3.17.0-0.1.master
```
Comment (ehelms) [inline packages/katello/katello/katello.spec:183]: Can you squash this changelog as well

File: packages/katello/katello/katello.spec (hunk 1 above)
Comment (jlsherrill) [reply packages/katello/katello/katello.spec:186]: hah, wasn't on latest rpm/develop

**Comment by jlsherrill** [comment]
I think this just needs to wait now on whatever magic generates a new gem from katello

**Comment by ehelms** [comment]
[test rpm]

**Pull request merged**